### PR TITLE
feat: add pattern option

### LINKED ISSUE #110: Provide minimatch patterns support for `src` option as `gh-pages` does
```js
{
    src: ['./foo/**/*', 'bar/*'],
    dst: '.'
    ...
}
```

### PR DESCRIPTION
**Pull request by dvirtz**
Closes #110

## Changes

add pattern option which is passed as gh-pages [src](https://github.com/tschaub/gh-pages?tab=readme-ov-file#optionssrc) option

- [x] New code is covered by tests
- [x] All the changes are mentioned in docs (readme.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,9 +15,9 @@ gh-pages publishing plugin for [semantic-release](https://github.com/semantic-re
 ## Install
 ```bash
 # yarn
-yarn add @qiwi/semantic-release-gh-pages-plugin --dev
+yarn add @wiki/semantic-release-gh-pages-plugin --dev
```
Comment (antongolub) [inline README.md:18]: I'm afraid, we're not ready to change the pkg name and scope )

**Review by antongolub — CHANGES_REQUESTED** [review]

**Review by dvirtz — COMMENTED** [review]
rebased to fix commit author

**Comment by antongolub** [comment]
@dvirtz,

Another rebase is required to fix the test

**Review by dvirtz — COMMENTED** [review]
> @dvirtz,
> 
> Yet another rebase is required to fix the test
> 
>  

done

**Review by antongolub — APPROVED** [review]
lgtm

**Pull request merged**